packages/types/src/util.ts

```diff
@@ -80,6 +80,7 @@ export interface RegionInfo {
   path?: string;
   signingService?: string;
   signingRegion?: string;
+  partition?: string;
```

Comment:
Should partition be required property in RegionInfo?
```suggestion
  partition: string;
```

Reply:
I don't feel strongly on this one. I made it optional just make the `regionInfoProvider` easier to mock.